ls1intum/Artemis
# `Exam mode`: Adjust checklist overview for teaching assistants

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


We encountered a problem with the checklist overview when a teaching assistant enters the exam management overview. Some data displayed there is fetched from an endpoint that was only accessible for at least instructors.
Omitting the whole checklist would be inordinate as the tutors would benefit from some of the information. On the other hand, the checklist includes some information the tutors should not necessarily have access to. Therefore I decided to adjust the checklist overview for tutors. The overview now displays the following information to the tutor:

- Exam conduction: Start and end date
- Exam correction: 
     - Correction round progress
     - Publish result date
     - Review state
     - Amount of resolved complaints

### Description

I adapted the corresponding endpoint that is returning the statistical data we need especially for the exam conduction part of the overview. It is now accessible for teaching assistants as well, but only includes the adjusted data described above to the response if a tutor sends the request. For instructors, everything should work as before. 
### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Tutor
- 1 Editor
- 1 Exam (feel free to try different states of the exam)

1. Log in to Artemis as Tutor
2. Navigate to Course Administration
3. Navigate to exam management. Make sure the checklist overview loads completely and no errors are shown in the console.
4. Check that the displayed information is correct by checking the checklist overview as instructor.
5. Log in as Editor and check the overview again. The view should be the same as for tutors.
### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage






### Screenshots


The checklist overview currently looks like this for tutors:
![image (6)](https://user-images.githubusercontent.com/80622272/155703615-916e8ca4-ead7-49fb-a046-36da3f7c2c08.png)

How it looks with this PR:
![image](https://user-images.githubusercontent.com/80622272/155891401-ab1176d8-b75b-4b2c-b359-17c22a86f47b.png)
